app/controllers/services_controller.rb

```diff
@@ -6,11 +6,12 @@ def index
 
     if name_query.present?
       services = services.where(Service.arel_table[:name].matches("%#{Service.sanitize_sql_like(name_query)}%"))
+      query_results_count = services.count
     end
 
     services = services.page(page).per(per_page)
 
-    render json: ServicesSerializer.new(services, total_services: true).attributes
+    render json: ServicesSerializer.new(services, total_services: query_results_count || true).attributes
   end
```

Comment:
i dunno if i'm oversimplifying this, but can we not just count the services before it's been paginated? like so:

```suggestion
  def index
    services = Service.order(name: :asc)

    if name_query.present?
      services = services.where(Service.arel_table[:name].matches("%#{Service.sanitize_sql_like(name_query)}%"))
    end

    total_services = services.count
    services = services.page(page).per(per_page)

    render json: ServicesSerializer.new(services, total_services: total_services).attributes
  end
```